Release 6.0.0-1

iarna opened the pull request:
[CHANGELOG](https://github.com/npm/npm/blob/release-next/CHANGELOG.md)

jdeniau committed:
publish: Better message when an "OTP" error occurs

PR-URL: https://github.com/npm/npm/pull/19580
Credit: @jdeniau
Reviewed-By: @iarna

iarna committed:
shrinkwrap: save requested ranges not versions

iarna committed:
audit: Add new audit command

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
npm-pick-manifest@2.1.0

iarna committed:
outdated: Make wanted respect latest interaction w/ semver

zkat committed:
tiny-relative-date@1.3.0

PR-URL: https://github.com/npm/npm/pull/19910
Credit: @zkat
Reviewed-By: @iarna

zkat committed:
cli-columns@3.1.2

PR-URL: https://github.com/npm/npm/pull/19910
Credit: @zkat
Reviewed-By: @iarna

zkat committed:
view: add humanized default view

PR-URL: https://github.com/npm/npm/pull/19910
Credit: @zkat
Reviewed-By: @iarna

zkat committed:
hook: add support for managing npm hooks

PR-URL: https://github.com/npm/npm/pull/20256
Credit: @zkat
Reviewed-By: @iarna

iarna committed:
install/deps: fix bug where optional status not saved to package-lock

Specifically, because we weren't adding optional deps to both `dependencies`
and `optionalDependencies` (the way `normalize-package-data` does), it meant
that `computeMetadata` didn't think the dep was actually required and as such
there was no way to compute its optional status.

iarna committed:
install: Filter optional deps during diff-trees

This allows `--no-optional` runs to include optional dependencies in the
lockfile while excluding them from installation on disk.

iarna committed:
is-only-optional: Read optional status from shrinkwrap when we have to

zkat committed:
inflate-shrinkwrap: allow fakeChildren for non-integrity types

Credit: @zkat
PR-URL: https://github.com/npm/npm/pull/20029
Reviewed-By: @iarna

iarna committed:
error-message: Fix messaging for 401 errors that lack a reason

Prevously they were falling through to 404 handling and being printed as:

npm ERR! 404 Registry returned 401 for DELETE on …

Credit: @iarna
Reviewed-By: @zkat
PR-URL: https://github.com/npm/npm/pull/20032

zkat committed:
docs: tiny changelog fixes

zkat committed:
byte-size@4.0.2

PR-URL: https://github.com/npm/npm/pull/19910
Credit: @zkat
Reviewed-By: @iarna

zkat committed:
shrinkwrap: suggest installing the merge driver (#20071)

PR-URL: https://github.com/npm/npm/pull/20071
Credit: @zkat
Reviewed-By: @iarna

zkat committed:
pack/publish: --dry-run, --json, and package previews (#19908)

PR-URL: https://github.com/npm/npm/pull/19908
Credit: @zkat
Reviewed-By: @iarna

zkat committed:
libnpmhook@3.0.0

PR-URL: https://github.com/npm/npm/pull/20256
Credit: @zkat
Reviewed-By: @iarna

zkat committed:
figgy-pudding@2.0.1

PR-URL: https://github.com/npm/npm/pull/20256
Credit: @zkat
Reviewed-By: @iarna

Commit:
feat: allow optional @ prefix on scope

Credit: @bcoe
Reviewed-By: @zkat
PR-URL: https://github.com/npm/npm/pull/20082

zkat committed:
cli: shinier update-notifier

This message is based on the one pnpm uses, which
I think takes care of all the false update messages and
missing -g that happens with the default notifier message.

It also tells people what "level" the update is.

Credit: @zkat
PR-URL: https://github.com/npm/npm/pull/20122
Reviewed-By: @iarna

SimenB committed:
npm: add npm cit command

PR-URL: https://github.com/npm/npm/pull/20126
Credit: @SimenB
Reviewed-By: @iarna

srl295 committed:
doc: note that package-lock version gets touched by npm-version

Credit: @srl295
PR-URL: https://github.com/npm/npm/pull/20140
Reviewed-By: @iarna

iarna committed:
unsupported: Add 10 to Node.js version list

iarna committed:
libcipm@1.6.1

Fixes bugs on docker and with some `prepare` scripts.

Fixes a bug where script hooks wouldn't be called.

Fixes a bug where `npm ci` and `--prefix` weren't compatible.

Credit: @isaacseymour
Credit: @umarov
Credit: @mikeshirov
Credit: @billjanitsch

Fixes: #19258

iarna committed:
lru-cache@4.1.2

Doc update.

Credit: @isaacs

iarna committed:
qrcode-terminal@0.12.0

Changes to the qrcode cli.

Credit: @kblum

iarna committed:
request@2.85.0

oauth changes

iarna committed:
marked@0.3.18

iarna committed:
tap@11.1.3

iarna committed:
worker-farm@1.6.0

Allow customizing childprocess.fork. Ditch 0.10 compat.

iarna committed:
validate-npm-package-license@3.0.3

Dependency version bumps.

Credit: @kemitchell

iarna committed:
tar@4.4.1

Switch to safe-buffer and Buffer.from

Credit: @isaacs
Credit: @ChALkeR

iarna committed:
ssri@5.3.0

Add option to throw when checks fail.

Credit: @zkat

iarna committed:
query-string@5.1.1

Allow partial imports

Credit: @Haraldson

iarna committed:
unsupported: Drop support for Node 4 & Node 7

iarna committed:
bin-links@1.1.2

Only rewrite the CR after a shebang (if any) when fixing up CR/LFs.
This saves us from mangling true binaries.

Credit: @iarna

iarna committed:
iferr@1.0.0

Update to require Node 6.

Credit: @shesek

iarna committed:
query-string@6.0.0

Require Node 6.

Credit: @sindresorhus

iarna committed:
is-cidr@2.0.5

New interface with renamed exports.

Credit: @silverwind

iarna committed:
standard@11.0.1

iarna committed:
standardize

iarna committed:
test: fix to support new bin-links behavior

iarna committed:
doc: update changelog for npm@5.9.0

iarna committed:
doc: retire npm@5 changelog to old majors section

iarna committed:
doc: update changelog for npm@6.0.0

zkat committed:
install: skip deprecated versions when possible

`npm-pick-manifest@2.1.0` included a change that makes it skip deprecated
versions when possible even if they would otherwise be the best semver range
match.

https://github.com/npm/npm/commit/3aaa6ef427b7a34ebc49cd656e188b5befc22bae
already uses `npm-pick-manifest` in such a way, and this patch guarantees that `pacote` itself
picks up this behavior as well (for `install`), since it defaults to including deprecated versions if
this option isn't passed through.

The patch for pacote to change this behavior can come later.

PR-URL: https://github.com/npm/npm/pull/20151
Credit: @zkat
Reviewed-By: @iarna

iarna committed:
6.0.0-next.0

iarna committed:
package-lock: Remove duplicate dependency

iarna committed:
iferr@1.0.0 (fill incompat deps)

iarna committed:
config: treat otps as strings to not eat leading zeros

Credit: @iarna

nwoltman committed:
version: Let version succeed when `package-lock.json` is gitignored

Credit: @nwoltman
PR-URL: https://github.com/npm/npm/pull/20154
Reviewed-By: @iarna
Fixes: #17168

buddydvd committed:
config: only create config dir before write

This change fixes the bug that creates
unnecessary ./etc folder when calling npm
with the --prefix flag.

PR-URL: https://github.com/npm/npm/pull/20212
Credit: @buddydvd
Reviewed-By: @iarna
Fixes: #17322

iarna committed:
has-modern-meta: Allow through bundled deps w/o modern meta

Credit: @iarna

iarna committed:
decompose-actions: Remove needless asynchrony

Credit: @iarna

iarna committed:
decompose-actions: Make duplicate actions impossible

Credit: @iarna

iarna committed:
decompose-actions: Ensure sources for bundled modules exist

Credit: @iarna

iarna committed:
diff-trees: Ensure all actions survive the sort

Credit: @iarna

iarna committed:
diff-trees: Only avoid adding bundled deps when they're the result of a link

Credit: @iarna

iarna committed:
npmignore: Exclude all tests from distribution

zkat committed:
cli: fix easter egg (#20214)

PR-URL: https://github.com/npm/npm/pull/20214
Credit: @zkat

zkat committed:
pack: add shasum and integrity info to preview

PR-URL: https://github.com/npm/npm/pull/20257
Credit: @zkat
Reviewed-By: @iarna

iarna committed:
shrinkwrap: Stop churning ranges back to versions in requires

Credit: @iarna

iarna committed:
shrinkwrap: Upgrade all requires to ranges

Credit: @iarna

iarna committed:
meta: Update npm's lockfile to ranges from versions

Credit: @iarna

iarna committed:
json-parse-better-errors@1.0.2

More friendly error messages.

PR-URL: https://github.com/zkat/json-parse-better-errors/pull/1
Credit: @Hoishin

iarna committed:
readable-stream@2.3.6

Credit: @mcollina

iarna committed:
update-notifier@2.4.0

Credit: @sindersorhus

iarna committed:
marked@0.3.19

Credit: @joshbruce

zkat committed:
test: use classic reporter on travis again

Credit: @zkat
PR-URL: https://github.com/npm/npm/pull/20276
Reviewed-By: @iarna

zkat committed:
node-gyp@3.6.2

Fixes: #20163
Credit: @zkat
PR-URL: https://github.com/npm/npm/pull/20276
Reviewed-By: @iarna

zkat committed:
test: verify that node-gyp stuff is correct

PR-URL: https://github.com/npm/npm/pull/20276
Credit: @zkat
Reviewed-By: @iarna

iarna committed:
perf: Fix timing catch

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
npm: Make --timing set loglevel=timing

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
inflate-shrinkwrap: Synthetic link deps get the right realpath

iarna committed:
test: Ensure optional status is saved to lock file

iarna committed:
npm: Upgrade package-lock

iarna committed:
libcipm@1.6.2

detect binding.gyp for default install lifecycle

PR-URL: https://github.com/zkat/cipm/pull/46
Fixes: https://github.com/zkat/cipm/issues/45
Credit: @caleblloyd
Reviewed-By: @zkat

iarna committed:
libnpx@10.1.1

add --always-spawn to opt out of process takeover optimization feature

Credit: @jdalton
PR-URL: https://github.com/zkat/npx/pull/172

iarna committed:
cacache@11.0.1

PR-URL: https://github.com/zkat/cacache/pull/128
Credit: @zkat

iarna committed:
figgy-pudding@3.1.0

Credit: @zkat

iarna committed:
pacote@8.0.0

Credit: @zkat

iarna committed:
ssri@6.0.0

Credit: @zkat

iarna committed:
retry@0.12.0

iarna committed:
libnpmhook@4.0.1

Credit: @zkat

iarna committed:
npm-package-arg@6.1.0

Credit: @zkat

iarna committed:
read-package-tree@5.2.1

Credit: @zkat

iarna committed:
doc: update changelog for npm@5.10.0

iarna committed:
doc: update changelog for npm@6.0.0

iarna committed:
test: Change bad url in test in anticipation of aliasing

Credit: @iarna

zkat commented on CHANGELOG.md:
```diff
@@ -1,1929 +1,267 @@
+  Allow packages with non-registry specifiers to follow the fast path that
+  the we use with the lock-file for registry specifiers. This will improve install time
+  especially when operating only on the package-lock (`--package-lock-only`).
 
```
extra blank line

zkat commented on CHANGELOG.md:
```diff
@@ -1,1929 +1,267 @@
-* [`1ca916a1e`](https://github.com/npm/npm/commit/1ca916a1e9cf94691d1ff2e5e9d0204cfaba39e1)
-  Fix bug where when switching from a linked module to a non-linked module, the dependencies
-  of the module wouldn't be installed on the first run of `npm install`.
+  Fix the a bug where `npm i --only=prod` could remove development
```
Missing `*` and shasum?

iarna replied:
Not as such… this was bug fix was drawn out of the other through a rather complicated series of circumstances. (The change to fakeChildren handling resulted in tests failing because of the `--only=prod` bug now showing because their children were now being inflated as fakeChildren. However the `--only=prod` bug as active for ANY dependency instantiated as a fakeChild. We just didn't have tests that hit that code path, due to most of the tests using `file:` dependencies.

iarna replied:
Arguably it could have been a separate commit. I ended up squashing it into the original as I was fixing test failures in the merged branch.

zkat commented on CHANGELOG.md:
```diff
@@ -1,1929 +1,267 @@
+  Improve messaging when two-factor authentication is required while publishing.
+  ([@jdeniau](https://github.com/jdeniau))
+* [`471ee1c5b`](https://github.com/npm/npm/commit/471ee1c5b58631fe2e936e32480f3f5ed6438536)
+* [`0da38b7b4`](https://github.com/npm/npm/commit/0da38b7b4aff0464c60ad12e0253fd389efd5086)
```
extra `*` here

zkat commented on CHANGELOG.md:
```diff
@@ -1,1929 +1,267 @@
-  `npm-lifecycle@2.0.1`
+* [`a9e722118`](https://github.com/npm/npm/commit/a9e7221181dc88e14820d0677acccf0648ac3c5a)
+  [#20256](https://github.com/npm/npm/pull/20256)
+  add support for managing npm hooks
```
Capitalisation and `.` missing

iarna replied:
oh good catch, this should actually say more as this is kinda exciting.

zkat commented on CHANGELOG.md:
```diff
@@ -1,1929 +1,267 @@
+* [`429498a8c`](https://github.com/npm/npm/commit/429498a8c8d4414bf242be6a3f3a08f9a2adcdf9)
+  [#20029](https://github.com/npm/npm/pull/20029)
+  Allow packages with non-registry specifiers to follow the fast path that
+  the we use with the lock-file for registry specifiers. This will improve install time
```
"..the fast path that <strike>the </strike>we use with the lock-file..."

zkat reviewed (approved):
Looks great! Some nitpick/style/typo-related comments but otherwise 👍 🐑 🚀 💥

iarna committed:
update AUTHORS

iarna committed:
6.0.0-next.1

iarna committed:
npm: Fixup bundling

zkat committed:
libnpx@10.2.0

zkat committed:
pacote@8.1.0

vipranarayan14 committed:
docs: Fix broken link to 'private-modules' page. (#20331)

PR-URL: https://github.com/npm/npm/pull/20331
Credit: @vipranarayan14
Reviewed-By: @zkat

aleclarson committed:
docs: add --if-present to run-script docs (#20279)

PR-URL: https://github.com/npm/npm/pull/20279
Credit: @aleclarson
Reviewed-By: @iarna

jdalton committed:
init: use npx for extended initializers (#20303)

PR-URL: https://github.com/npm/npm/pull/20303
Credit: @jdalton
Reviewed-By: @zkat

zkat committed:
doc: wordsmith new npm init docs a bit

raymondfeng committed:
install-test: fix shrinkwrap handling of package-lock.json (#20358)

For a simple project with .npmrc containing 'package-lock=false',
`npm install-test` command will generate package-lock.json.*.

If there is an existing package-lock.json, npm install-test triggers
lifecycle actions twice for both install and test.

PR-URL: https://github.com/npm/npm/pull/20358
Credit: @raymondfeng
Reviewed-By: @zkat

jdalton committed:
test: ensure npm init forwards arguments (#20372)

PR-URL: https://github.com/npm/npm/pull/20372
Credit: @jdalton
Reviewed-By: @zkat

jdalton committed:
docs: add --scope to npm init usage (#20373)

PR-URL: https://github.com/npm/npm/pull/20373
Credit: @jdalton
Reviewed-By: @zkat

iarna committed:
audit: Temporarily suppress git metadata till there's an opt-in

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
npm-registry-fetch@1.1.0

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
npm-audit-report@1.0.5

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
audit: Add docs

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

zkat committed:
pkglock: add from field back into git dependencies (#20384)

PR-URL: https://github.com/npm/npm/pull/20384
Credit: @zkat
Reviewed-By: @iarna

iarna committed:
test: Default audit to off when testing

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
audit: Avoid config-meta's literal-only test

These are about the only legitimate uses of dynamic config lookups.

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
test: JSON parse error message changed slightly

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
test: standard common-tap

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

iarna committed:
audit: Ensure we don't mutate the shrinkwrap

PR-URL: https://github.com/npm/npm/pull/20389
Credit: @iarna
Reviewed-By: @zkat

zkat committed:
doc: update changelog for npm@6.0.0

zkat commented:
this was merged in its 6.0.0-next.1 form already

jdalton commented:
:tada::tada::tada::tada::tada::tada:
